src/esp/physics/PhysicsManager.cpp

```diff
@@ -92,12 +93,14 @@ int PhysicsManager::addObject(const std::string& configFile,
   return physObjectID;
 }
 
-int PhysicsManager::removeObject(const int physObjectID) {
+int PhysicsManager::removeObject(const int physObjectID, bool deleteSceneNode) {
```

Comment:
Why is deleteSceneNode needed? It seems to always be true.

Reply:
Also, why is the `pysObjectID` returned? I don't see any purpose in that.

Reply:
I tried to figure out a better variant of the API (like returning the `objectNode` so the caller can delete it or not), but it's not ideal either, as it changes the default to "not delete".

Reply:
I agree the ID return is unnecessary, I'll change this. 

I provide the option for SceneNode delete so that the object can be detached from physics while preserving the SceneNode for rendering if desired. Most of the time you'll want to remove it, but this provides an alternative (maybe you are holding a reference to the node for some purpose).